packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -131,10 +149,17 @@ export interface MachineImageConfig {
  * on the instance if you are using this image.
  *
  * The AMI ID is selected using the values published to the SSM parameter store.
+ *
+ * @deprecated Use `MachineImage.fromSsmParameter()` instead
```

Comment:
```suggestion
 * @deprecated Use `GenericSsmParameterImage` instead
```

Reply:
Oh no. I don't want people using the class directly, I want them to use the factories.